d365fo.tools/functions/invoke-d365dbsyncmodule.ps1

```diff
@@ -103,43 +119,37 @@ function Invoke-D365DbSyncModule {
 
     begin {
         Invoke-TimeSignal -Start
-
-        $modules = @()
     }
 
     process {
-        if ($Module -is [array] -and $Module.Count -gt 1) {
-            $modules = $Module
-        }
-        else {
-            $modules += $Module
+        foreach ($moduleLocal in $Module) {
+            # Retrieve all sync elements of provided module name
+            $allModelSyncElements = $moduleLocal | Get-SyncElements
+
+            # Build parameters for the partial sync function
+            $syncParams = @{
+                SyncList             = $allModelSyncElements.BaseSyncElements;
+                SyncExtensionsList   = $allModelSyncElements.ExtensionSyncElements;
+                Verbosity            = $Verbosity;
+                BinDirTools          = $BinDirTools;
+                MetadataDir          = $MetadataDir;
+                DatabaseServer       = $DatabaseServer;
+                DatabaseName         = $DatabaseName;
+                SqlUser              = $SqlUser;
+                SqlPwd               = $SqlPwd;
+                ShowOriginalProgress = $ShowOriginalProgress;
+                OutputCommandOnly    = $OutputCommandOnly
+            }
+
+            # Call the partial sync using required parameters
+            $resSyncModule = Invoke-D365DBSyncPartial @syncParams
+
+            $resSyncModule
         }
     }
 
     end {
-        # Retrieve all sync elements of provided module name
-        $allModelSyncElements = $modules | Get-SyncElements
-
-        # Build parameters for the partial sync function
-        $syncParams = @{
-            SyncList=$allModelSyncElements.BaseSyncElements;
-            SyncExtensionsList = $allModelSyncElements.ExtensionSyncElements;
-            Verbosity = $Verbosity;
-            BinDirTools=$BinDirTools;
-            MetadataDir=$MetadataDir;
-            DatabaseServer=$DatabaseServer;
-            DatabaseName=$DatabaseName;
-            SqlUser=$SqlUser;
-            SqlPwd=$SqlPwd;
-            ShowOriginalProgress=$ShowOriginalProgress;
-            OutputCommandOnly=$OutputCommandOnly
-        }
-
-        # Call the partial sync using required parameters
-        $resSyncModule = Invoke-D365DBSyncPartial @syncParams
-
-        $resSyncModule
-
+        
```

Comment:
The reason I implemented it the way i did was for performance. Collecting all of the objects that need to be synced up front and then performing the sync means that the overhead of running the sync (regardless of the number of objects) only needs to be run once. If we run the sync in the process section, then the sync logic will be called for each module that is passed in (whether by array or by pipe).

Reply:
Point taken!

I'll revert my changes, and make a note about it in the help section. 

So if you want reports per module, then you will suffer from a performance point of view.

And if you just want to make it is fast as possible, you will have to live with the fact that the output is mixed across your models / modules.

Reply:
From my point of view, I would rather have the performance. If there is an error during the sync, I just want to know what object caused the error, regardless of the module it was in.

Reply:
Please give it a look now, and pull it so we can test it works for you.